library/src/options.cxx

```diff
@@ -104,15 +104,12 @@ class options::internals
     }
     catch (const std::bad_variant_access&)
     {
-      std::string error = "Trying to get option reference " + name + " with incompatible type";
-      log::error(error);
-      throw options::incompatible_exception(error + "\n");
+      throw options::incompatible_exception(
+        "Trying to get option reference " + name + " with incompatible type");
     }
     catch (const std::out_of_range&)
     {
-      std::string error = "Option " + name + " does not exist";
-      log::error(error);
-      throw options::inexistent_exception(error + "\n");
+      throw options::inexistent_exception("Option " + name + " does not exist");
```

Comment:
why no log anymore ?

Reply:
I removed it because we want to get a value without knowing its type from python, so we just `try` until we find the right one and it would spam the logging.

More generally, logging when throwing feels wrong, either you'll catch the exception somewhere and you should log there and then, or you're not catching it and logging won't help much when it runs all the way up and the program is terminated.

Reply:
I agree with @snoyer  
It's `log` or `throw`, but doing both is a bad design in my opinion.

Reply:
got it